core/src/main/java/com/yahoo/imapnio/async/data/ListStatus.java

```diff
@@ -0,0 +1,53 @@
+package com.yahoo.imapnio.async.data;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import com.sun.mail.imap.protocol.ListInfo;
+import com.sun.mail.imap.protocol.Status;
+
+/**
+ * This class contains a ListInfo and Status object. It is used for storing the result from LIST-STATUS command. RFC5819 states below.
+ * <p>
+ * For each selectable mailbox matching the list pattern and selection options, the server MUST return an untagged LIST response followed by an
+ * untagged STATUS response containing the information requested in the STATUS return option. If an attempted STATUS for a listed mailbox fails
+ * because the mailbox can't be selected , the STATUS response MUST NOT be returned and the LIST response MUST include the \NoSelect attribute.
+ * </p>
+ */
+public class ListStatus {
```

Comment:
How much need do you have for combining list response of a folder with status?

Reply:
Since this is mandated by RFC, we resolve my own comment.

Reply:
lets fix to return a list of listinfo and a list or map of status.

Reply:
done.